### PR TITLE
Unit Test Coaching Dashboard

### PR DESCRIPTION
**Pull request by LuckyHariharan**
Created skillifyUserTest.test.ts which contains 2 tests: a test for an unchanged initial state and a test state with test data. Both test pass. 
image

Additional commits from my [Coaching Dashboard Redux PR](https://github.com/skillify-ca/skillify-web/pull/831) are in this PR because this branch (LH-CoachingDashboardUnitTest) was created based on the Redux branch.

### CODE REVIEW

File: components/coding/studentPortal/ProfileDetailCard.tsx [hunk 1]
```diff
@@ -31,11 +31,19 @@ function ProfileDetailCard({
         <p className=" font-bold "> {name}</p>
       </div>
       <div className="flex flex-col ml-4 mb-4 gap-4 items-start">
-        <JoinedDateComponent user={name} createdAt={joinDate} />
+        <JoinedDateComponent
+          user={name}
+          textSize={"small"}
```
Comment (LuckyHariharan) [inline components/coding/studentPortal/ProfileDetailCard.tsx:36]: these changes are from the redux branch that this branch was based off of. I was missing the textSize={"small"} prop which I added and reformatted.

File: components/coding/studentPortal/ProfileDetailCard.tsx [hunk 2]
```diff
@@ -31,11 +31,19 @@ function ProfileDetailCard({
...
+          createdAt={joinDate}
+        />
 
-        <BadgesDisplayedComponent earnedBadges={badges} totalBadges={50} />
-        <p className="underline ">current focus: {currentFocus}</p>
-        <p className="underline">next goal: {nextGoal}</p>
+        <BadgesDisplayedComponent
+          earnedBadges={badges}
+          textSize={"small"}
```
Comment (LuckyHariharan) [inline components/coding/studentPortal/ProfileDetailCard.tsx:42]: again added this to fix the missingthe textSize={"small"} prop.

File: redux/skillifyUsersSlice.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export const skillifyUsersSlice: Slice = createSlice({
+  name: "skillifyUsers",
+  initialState,
+  reducers: {
+    setUserList: (
+      state: SkillifyUsersState,
+      action: PayloadAction<Users[]>
+    ) => {
+      if (action.type == "skillifyUsers/setUserList") {
+        return { userList: [...action.payload] };
```
Comment (LuckyHariharan) [inline redux/skillifyUsersSlice.ts:22]: Added return statement for testing to work as expected.

File: components/coding/studentPortal/ProfileDetailCard.tsx (hunk 1 above)
Comment (vithushan19) [reply components/coding/studentPortal/ProfileDetailCard.tsx:36]: Great job, leaving that level of context!

File: redux/rootReducer.ts [hunk 4]
```diff
@@ -16,13 +18,15 @@ type State = {
   skillRatingsState: SkillRatingsState;
   userGoalsState: UserGoalsState;
   profileState: ProfileState;
+  skillifyUsersState: SkillifyUsersState;
```
Comment (vithushan19) [inline redux/rootReducer.ts]: Could this just be userState? Feels more in line with the naming elsewhere in the app

**Review by vithushan19 — APPROVED** [review]
Looking good to me

File: redux/rootReducer.ts (hunk 4 above)
Comment (LuckyHariharan) [reply redux/rootReducer.ts]: Oh kk sounds good.

File: redux/skillifyUsersSlice.ts [hunk 5]
```diff
@@ -1,4 +1,6 @@
 import { createSlice, PayloadAction, Slice } from "@reduxjs/toolkit";
+import { Action } from "lottie-react";
```
Comment (iamnotreddy) [inline redux/skillifyUsersSlice.ts:2]: this shouldn't be in here

**Pull request merged**